shopfloor_reception/services/reception.py

```diff
@@ -0,0 +1,1155 @@
+            "new_package_name": new_package_name,
+        }
+        return self._response(
+            next_state="confirm_new_package", data=data, message=message
+        )
+
+    def _select_document_from_product(self, product):
+        """Select the document by product
+
+        next states:
+            - set_lot: a single picking has been found for this packaging
+            - select_document: A single or no pickings has been found for this packaging
+        """
+        move_lines = self._move_line_by_product(product).filtered(
+            lambda l: l.picking_id.picking_type_id.id in self.picking_types.ids
+        )
+        pickings = move_lines.mapped("move_id.picking_id")
+        if len(pickings) == 1:
+            self._assign_user_to_picking(pickings)
+            if product.tracking not in ("lot", "serial"):
+                return self._response_for_set_quantity(pickings, move_lines)
+            return self._response_for_set_lot(pickings, move_lines)
+        elif len(pickings) > 1:
+            return self._response_for_select_document(
+                pickings=pickings,
+                message=self.msg_store.multiple_picks_found_select_manually(),
+            )
+        # If no available picking with the right state has been found,
+        # return an error
+        return self._response_for_select_document(
+            message=self.msg_store.product_not_found_in_pickings()
+        )
+
+    def _select_document_from_packaging(self, packaging):
+        """Select the document by packaging
+
+        next states:
+            - set_lot: a single picking has been found for this packaging
+            - select_document: A single or no pickings has been found for this packaging
+        """
+        move_lines = self._move_line_by_packaging(packaging).filtered(
+            lambda l: l.picking_id.picking_type_id.id in self.picking_types.ids
+        )
+        pickings = move_lines.mapped("move_id.picking_id")
+        if len(pickings) == 1:
+            self._assign_user_to_picking(pickings)
+            if packaging.product_id.tracking not in ("lot", "serial"):
+                return self._response_for_set_quantity(pickings, move_lines)
+            return self._response_for_set_lot(pickings, move_lines)
+        elif len(pickings) > 1:
+            return self._response_for_select_document(
+                pickings=pickings,
+                message=self.msg_store.multiple_picks_found_select_manually(),
+            )
+        # If no available picking with the right state has been found,
+        # return a barcode not found error message
+        return self._response_for_select_document(
+            message=self.msg_store.no_transfer_for_packaging(),
+        )
+
+    def _select_line_from_product(self, picking, product):
+        line = fields.first(
+            picking.move_line_ids.filtered(
+                lambda l: l.product_id == product and not l.result_package_id
```

Comment:
and shopfloor user is False or self.env.uid
+ allow to optionally show shopfloor user on the move line